browser/misc_metrics/page_metrics_tab_helper.cc

```diff
@@ -25,13 +28,33 @@ PageMetricsTabHelper::PageMetricsTabHelper(content::WebContents* web_contents)
 
 PageMetricsTabHelper::~PageMetricsTabHelper() = default;
 
+void PageMetricsTabHelper::DidStartNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (!CheckNavigationEvent(navigation_handle, false)) {
+    return;
+  }
+
+  last_started_host_ = navigation_handle->GetURL().host();
+  was_http_allowlist_ = IsHttpAllowedForHost(navigation_handle);
+}
+
+void PageMetricsTabHelper::DidRedirectNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (navigation_handle->GetURL().host() == last_started_host_) {
```

Comment:
can we just look at the the redirect chain from the navigation handle in `DidFinish` ? If this thing only maintains last_started_host_ etc